lib/Elastica/Transport/HttpAsync.php

```diff
@@ -0,0 +1,103 @@
+<?php
+namespace Elastica\Transport;
+
+use Elastica\Request;
+use Elastica\Response;
+use Elastica\Util;
+
+/**
+ * Elastica HttpAsync Transport object.
+ *
+ * @author Norberto Gomez
+ */
+class HttpAsync extends AbstractTransport
+{
+    /**
+     * Http scheme.
+     *
+     * @var string Http scheme
+     */
+    protected $_scheme = 'http';
+
+    /**
+     * Curl resource to reuse.
+     *
+     * @var resource Curl resource to reuse
+     */
+    protected static $_curlConnection;
+
+    /**
+     * Makes calls to the elasticsearch server.
+     *
+     * All calls that are made to the server are done through this function
+     *
+     * @param \Elastica\Request $request
+     * @param array             $params  Host, Port, ...
+     *
+     * @return \Elastica\Response Response object
+     */
+    public function exec(Request $request, array $params)
+    {
+        $connection = $this->getConnection();
+        $baseUri = $this->_buildBaseUri($request, $connection);
+        $command = 'curl';
+        $command_params = [];
+        $command_params[] = ' -m ' . $connection->getTimeout();
+        $data = $request->getData();
+
+        $headers = [];
+        array_push($headers, sprintf('Content-Type: %s', $request->getContentType()));
+
+        foreach ($headers as $header) {
+            $command_params[] = ' -H ' . '"' . $header . '"';
+        }
+
+        if (!empty($data) || '0' === $data) {
+            $content = str_replace('\/', '/', $data);
+            $command_params[] = ' --data ' . "'$content
+'";
+        }
+
+        $curl_command = 'nohup ' . $command . implode(' ', $command_params) . ' ' . $baseUri . ' > /dev/null';
+
+        exec($curl_command);
```

Comment:
You might want to note here that even if the exec fails, the async is considered a success. We're almost doing this UDP style :D 

You could use `shell_exec()` and check the output for a successful response, and return that in the `Response` object. Maybe the result gives you the query?

Also if this is just for cURL, do we provide a timeout? Maybe a timeout constant might be useful, or even a configurable value...

Reply:
Hey, that's the point, as discussed with Jesus, we are just sending the cURL call and forget about the response, since we are only sending one call, for each register in the Elasticsearch index. otherwise it will not remain async since we have to wait for the response to continue.

And yes we provide a timeout in `$command_params[] = ' -m ' . $connection->getTimeout();` by default in `Elastica` if we don't provide is `300`

Reply:
Ah true, with the `nohup`, you're right!